main/modes/games/pango/pango.c

```diff
@@ -1546,4 +1587,384 @@ void pa_advanceToNextLevelOrGameClear(pango_t* self)
     {
         pangoSaveUnlockables(self);
     }
+}
+
+void updateBtnStateAttractMode(pango_t* self)
+{
+    if (!(self->gameData.frameCount % 16))
+    {
+        if ((esp_random() % 10) > 6)
+        {
+            self->gameData.btnState &= ~15;
+            self->gameData.btnState
+                |= (((esp_random() % 2) << (esp_random() % 2)) + ((esp_random() % 2) << 2 << (esp_random() % 2)));
+        }
+
+        if ((esp_random() % 10) > 8)
+        {
+            self->gameData.btnState |= PB_A;
+        }
+        else
+        {
+            self->gameData.btnState &= ~PB_A;
+        }
+    }
+}
+
+/*
+uint16_t getRandomDirection(){
+    switch(esp_random() % 8){
+        case 0:
+            return PA_DIRECTION_NORTH;
+        case 1:
+            return PA_DIRECTION_SOUTH;
+        case 2:
+            return PA_DIRECTION_WEST;
+        case 3:
+            return PA_DIRECTION_EAST
+    }
+}
+*/
+
+void changeStateDemoControls(pango_t* self)
+{
+    pa_deactivateAllEntities(&(self->entityManager), false);
+    pango->entityManager.activeEnemies = 0;
+    pa_loadMapFromFile(&(pango->tilemap), "demo01.bin");
+    pa_setDifficultyLevel(&(pango->wsgManager), &(pango->gameData), 2);
+    self->entityManager.playerEntity
+        = pa_createPlayer(&(self->entityManager), (0 << PA_TILE_SIZE_IN_POWERS_OF_2) + PA_HALF_TILE_SIZE,
+                          (7 << PA_TILE_SIZE_IN_POWERS_OF_2) + PA_HALF_TILE_SIZE);
+    self->menuState     = 0;
+    self->frameTimer    = 0;
+    self->menuSelection = 1;
+
+    pango->update = &updateDemoControls;
+}
+
+void updateDemoControls(pango_t* self, int64_t elapsedUs)
+{
+    if (self->btnState & PB_START)
+    {
+        switch (self->gameData.gameState)
+        {
+            case PA_ST_TITLE_SCREEN:
+            default:
+                pangoChangeStateMainMenu(self);
+                break;
+            case PA_ST_READY_SCREEN:
+                pa_initializeGameDataFromTitleScreen(&(self->gameData));
+                self->entityManager.activeEnemies = 0;
+                pa_setDifficultyLevel(&(self->wsgManager), &(self->gameData), self->gameData.level);
+                pa_loadMapFromFile(&(self->tilemap), "preset.bin");
+                pa_generateMaze(&(self->tilemap));
+                pa_placeEnemySpawns(&(self->tilemap));
+
+                changeStateReadyScreen(self);
+                return;
+                break;
```

Comment:
You don't need `break` after `return`, but no harm no foul

Reply:
I'd rather overinclude `break` rather than forget one and spend an hour to realize that one is missing  😉